### PR TITLE
Shop and Cart Items

### PR DESCRIPTION
**Pull request by FrankOelsch**
All crud operations

### CODE REVIEW

**Review by Kheirah — APPROVED** [review]
Very good work!

There are some adjustments you can make to improve the quality of your code. Have a look, make some changes, and you're good to go! 🥳

File: src/components/Item.js [hunk 1]
```diff
@@ -1,27 +1,198 @@
...
+              data-for="ttdelete"
+              onClick={() => onDelete(item.id)}
+            ></Delete>
+            <ReactTooltip id="ttdelete" type="info">
+              Löschen
+            </ReactTooltip>
+          </div>
+        )}
+
+        {item.inCart || (
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/components/Item.js:43]: Thanks!

File: src/pages/Cart.js [hunk 2]
```diff
@@ -1,124 +1,191 @@
...
+  const filteredCartItems = cartItems.filter((item) => {
+    return item.for === config.system || item.for === "all";
   });
 
-  const filterArray = search(searchString, falseArray, {
-    keySelector: (obj) => obj.name,
-  });
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+
+  const [cartItem, setCartItem] = useState({ name: "blub", id: "0" });
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:57]: done

File: src/pages/Cart.js [hunk 3]
```diff
@@ -1,124 +1,191 @@
...
+
+      if (newArray) {
+        const newItem = newArray[0];
+
+        setCartItems((cartItems) => [
+          ...cartItems,
+          { ...newItem, inCart: !newItem.inCart },
+        ]);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:86]: done

File: src/pages/Cart.js [hunk 4]
```diff
@@ -1,124 +1,191 @@
...
+      const newArray = shopItems.filter((item) => {
+        return item.id === id;
+      });
+
+      if (newArray) {
+        const newItem = newArray[0];
+
+        setCartItems((cartItems) => [
+          ...cartItems,
+          { ...newItem, inCart: !newItem.inCart },
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:83]: done

File: src/pages/Cart.js [hunk 5]
```diff
@@ -1,124 +1,191 @@
...
+      })
+    );
+  }
+
+  function handleEdit(id) {
+    const newArray = cartItems.filter((item) => {
+      return item.id === id;
+    });
+
+    if (newArray) {
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:102]: done

**Pull request merged**